frb_codegen/src/source_graph.rs

```diff
@@ -282,53 +287,81 @@ impl Module {
-                                file_path,
-                                module_path,
-                                source,
-                                scope: None,
-                            };
+                                tried.push(file_path);
+                                let file_path = folder_path.join(&module_name).join("mod.rs");
+                                if file_path.exists() {
+                                    return Ok(file_path);
+                                }
 
-                            if file_exists {
-                                child_module.resolve();
+                                tried.push(file_path);
+                                Err(tried)
                             }
 
-                            child_module
+                            let file_path =
+                                get_module_file_path(ident.to_string(), &self.file_path);
+
+                            match file_path {
+                                Ok(file_path) => {
+                                    let source = {
+                                        let source_rust_content =
+                                            fs::read_to_string(&file_path).unwrap();
+                                        debug!("Trying to parse {:?}", file_path);
+                                        Some(ModuleSource::File(
+                                            syn::parse_file(&source_rust_content).unwrap(),
+                                        ))
+                                    };
+                                    let mut child_module = Module {
+                                        visibility: syn_vis_to_visibility(&item_mod.vis),
+                                        file_path,
+                                        module_path,
+                                        source,
+                                        scope: None,
+                                    };
+
+                                    child_module.resolve();
+                                    child_module
+                                }
+                                Err(tried) => {
+                                    warn!(
+                                        "Skipping unresolvable module {} (tried {})",
+                                        &ident,
+                                        tried
+                                            .into_iter()
+                                            .map(|it| it.to_string_lossy().to_string())
+                                            .fold(String::new(), |mut a, b| {
+                                                a.push_str(&b);
+                                                a.push(',');
+                                                a
+                                            })
+                                    );
+                                    continue;
+                                }
+                            }
```

Comment:
There seems to be not tests yet, so we cannot verify the behavior. What about we add files like `some_module.rs`, `some_module/sub_module.rs`, and a function that uses the structs in those modules in `api.rs` and a verify test in `.dart`. Then we are pretty sure this PR works :)